### PR TITLE
types/json: remove old JSON type.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@jackysp PTAL

File: types/json/binary_functions.go [hunk 1]
```diff
@@ -72,6 +76,71 @@ func (bj BinaryJSON) Unquote() (string, error) {
...
+// According RFC 3629, the max length of utf8 characters is 4 bytes.
+// And MySQL use 4 bytes to represent the unicode which must be in [0, 65536).
+func decodeEscapedUnicode(s []byte) (char [4]byte, size int, err error) {
+	size, err = hex.Decode(char[0:2], s)
+	if err != nil || size != 2 {
+		// The unicode must can be represented in 2 bytes.
+		return char, 0, errors.Trace(err)
+	}
+	var unicode uint16
+	err = binary.Read(bytes.NewReader(char[0:2]), binary.BigEndian, &unicode)
```
Comment (jackysp) [inline types/json/binary_functions.go]: I know it is an old JSON implement. I just want to know is there a test case cover this?

Comment (coocood) [reply types/json/binary_functions.go]: There are `JSONUnquote` tests in this package and `expression`.

**Pull request merged**